bluesky/plans.py

```diff
@@ -92,9 +92,12 @@ def _pre(self):
         # itself as its argument and returns a generator of messages
         if self.pre_run is not None:
             yield from self.pre_run(self)
+        staged = []  # to track duplicates
```

Comment:
"But Dan," you ask, "why aren't you just using `set()`?" Because it is good to stage each category of objects in a deterministic order. There is no `OrderedSet` in the standard lib.